### PR TITLE
Docker Continuous Deployment creation

### PR DESCRIPTION
**Pull request by David-mrc**
on publish of a new release, creates an artifact and loads the app.jar into a docker image, and uploads the image to docker hub

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          name: Package
+          path: .
+  
+  build-docker-image:
+    name: Publish to Docker Hub
+    runs-on: ubuntu-latest
+    needs: publish
+    steps:
+      - uses: actions/checkout@v2
```
Comment (dcs-b) [inline .github/workflows/publish.yml]: We should use actions/checkout@v3 instead of v2

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      packages: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - run: mvn -B package --file pom.xml -DskipTests
+      - run: mkdir staging && cp target/*.jar staging
+      - uses: actions/upload-artifact@v1
```
Comment (dcs-b) [inline .github/workflows/publish.yml]: We should use actions/upload-artifact@v3 instead of v1

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Publish package to GitHub Packages
```
Comment (dcs-b) [inline .github/workflows/publish.yml]: We can rename it to CD and add all CD jobs in this workflow and then rename the file to CD.yml to stay consistent.

**Review by dcs-b — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (David-mrc) [reply .github/workflows/publish.yml]: ok

File: .github/workflows/publish.yml (hunk 2 above)
Comment (David-mrc) [reply .github/workflows/publish.yml]: okok

File: .github/workflows/publish.yml (hunk 3 above)
Comment (David-mrc) [reply .github/workflows/publish.yml]: okokok

**Review by dcs-b — APPROVED** [review]
Perfect. Solved.

**Pull request merged**